src/media/media.service.spec.ts

```diff
@@ -0,0 +1,119 @@
+  const clientProxy = {
+    emit: jest.fn().mockImplementation(() => ({
+      toPromise: jest.fn(),
+    })),
+  };
+
+  beforeEach(async () => {
+    const moduleRef: TestingModule = await Test.createTestingModule({
+      imports: [ConfigModule],
+      providers: [
+        MediaService,
+        MediaRepository,
+        { provide: 'RABBITMQ_BROKER', useValue: clientProxy },
+      ],
+    }).compile();
+
+    mediaService = moduleRef.get<MediaService>(MediaService);
+    mediaRepo = moduleRef.get<MediaRepository>(MediaRepository);
+  });
+
+  it('Should be defined and have the necessary methods', () => {
+    // assertions
+    expect(mediaService).toBeDefined();
+    expect(mediaService).toHaveProperty('uploadService');
+  });
+
+  describe('uploadService method', () => {
+    it('Should return undefined', async () => {
+      // data
+      const file = {
+        buffer: {} as Buffer,
+        destination: '',
+        fieldname: '',
+        filename: '',
+        mimetype: '',
+        originalname: '',
+        path: '',
+        size: 10,
+        stream: {} as Readable,
+        encoding: '',
+      };
+      const createdFileInDB = {
+        uuid: 'test-uuid-of-file-after-it-is-added-in-DB',
+      };
+
+      const uploadData: MediaUploadDTO = {
+        entity_id: 'entity-test-id',
+        entity_type: 'option',
+      };
+
+      // mocks
+      mediaRepo.addFile = jest.fn().mockResolvedValueOnce(createdFileInDB);
+
+      // actions
+      const res = await mediaService.uploadService(file, uploadData);
+
+      // assertions
+      expect(res).toBeUndefined();
+    });
+
+    it('Should call rabbitMQClient with necessary values to emit the message', async () => {
+      // data
+      const file = {
+        buffer: {} as Buffer,
+        destination: '',
+        fieldname: '',
+        filename: '',
+        mimetype: '',
+        originalname: '',
+        path: '',
+        size: 10,
+        stream: {} as Readable,
+        encoding: '',
+      };
+      const createdFileInDB = {
+        uuid: 'test-uuid-of-file-after-it-is-added-in-DB',
+      };
+
+      const uploadData: MediaUploadDTO = {
+        entity_id: 'entity-test-id',
+        entity_type: 'option',
+      };
+
+      // mocks
+      mediaRepo.addFile = jest.fn().mockResolvedValueOnce(createdFileInDB);
+
+      // actions
+      await mediaService.uploadService(file, uploadData);
+
+      // assertions
+      expect(clientProxy.emit).toHaveBeenCalledWith('added_media_data', {
+        file_url: `https://undefined/api/media/${createdFileInDB.uuid}`,
+        entity_type: uploadData.entity_type,
+        entity_id: uploadData.entity_id,
+      });
+    });
+
+    it('Should throw error if file not found', () => {
+      // mocks
+      mediaRepo.addFile = jest.fn().mockResolvedValueOnce(undefined);
+
+      // assertions
+      expect(mediaService.uploadService).rejects.toThrowError(
+        new NotFoundException('File not found'),
+      );
+    });
```

Comment:
I've sent a request using Postman, where I only included the image file, and did not include the entity_id and entity_option, and the request was sent successfully and a response {ack: true} was received.

Shouldn't this be tested also that the required data must be present?

Reply:
yes, there was a bug in code and the validations weren't running successfully. All validations were fixed